contracts/HATKlerosConnector.sol

```diff
@@ -0,0 +1,490 @@
+// SPDX-License-Identifier: MIT
+
+/**
+ *  @authors: [@unknownunknown1]
+ *  @reviewers: []
+ *  @auditors: []
+ *  @bounties: []
+ *  @deployments: []
+ */
+
+pragma solidity 0.8.16;
+
+import {IDisputeResolver, IArbitrator} from "@kleros/dispute-resolver-interface-contract/contracts/IDisputeResolver.sol";
+import "./libraries/CappedMath.sol";
+import "./HATArbitrator.sol"; // TODO: add interface
+import "./interfaces/IHATVault.sol";
+
+/**
+ *  @title HATKlerosConnector
+ *  @dev This contract acts a connector between HatsFinance and Kleros court.
+ *  This contract trusts that the Arbitrator is honest and will not reenter or modify its costs during a call.
+ *  The arbitrator must support appeal period.
+ *  The contract also trusts that HATArbitrator contract is honest and won't reenter.
+ */
+contract HATKlerosConnector is IDisputeResolver {
+    using CappedMath for uint256;
+
+    uint256 private constant RULING_OPTIONS = 2; // The amount of non 0 choices the arbitrator can give.
+    uint256 private constant MULTIPLIER_DIVISOR = 10000; // Divisor parameter for multipliers.
+  
+    enum Decision {
+        None, // Court wasn't able to make a decisive ruling. In this case the claim is executed. Both sides will get their appeal deposits back in this case.
+        ExecuteResolution, // Accept the claim as it is without changing.
+        DismissResolution // Dismiss the claim.
+    }
+
+    struct DisputeStruct {
+        bytes32 claimId; // Id of the claim in HATVault contract.
+        uint256 externalDisputeId; // Id of the dispute created in Kleros court.
+        Decision ruling; // Ruling given by the arbitrator.
+        bool resolved; // True if the dispute has been resolved.
+        Round[] rounds; // Appeal rounds.
+    }
+
+    // Round struct stores the contributions made to particular sides.
+    // - 0 side for `Decision.None`.
+    // - 1 side for `Decision.ExecuteResolution`.
+    // - 2 side for `Decision.DismissResolution`.
+    struct Round {
+        uint256[3] paidFees; // Tracks the fees paid in this round in the form paidFees[side].
+        bool[3] hasPaid; // True if the fees for this particular side have been fully paid in the form hasPaid[side].
+        mapping(address => uint256[3]) contributions; // Maps contributors to their contributions for each side in the form contributions[address][side].
+        uint256 feeRewards; // Sum of reimbursable appeal fees available to the parties that made contributions to the side that ultimately wins a dispute.
+        uint256[] fundedSides; // Stores the sides that are fully funded.
+    }
+
+    address public immutable governor; // Governor of this contract.
+    IArbitrator public immutable klerosArbitrator; // The kleros arbitrator contract (e.g. Kleros Court).
+    HATArbitrator public immutable hatArbitrator; // Address of the Hat arbitrator contract.
+    IHATVault public immutable vault; // Vault contract.
```

Comment:
the HATArbitrator contract is written so it can be the arbitrator for different vaults. So this should not be a state variable, but passed as a parameter to the various functions